lib/pbench/cli/agent/commands/generate_token.py

```diff
@@ -0,0 +1,71 @@
+"""pbench-generate-token"""
+
+import click
+import json
+import requests
+
+from pbench.agent.base import BaseCommand
+from pbench.cli.agent import pass_cli_context
+from pbench.cli.agent.options import common_options
+
+
+class GenerateToken(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
+
+    def execute(self):
+        """Generate a token used to access the pbench server RESTful API"""
+
+        # FIXME:  Should we be logging anything here?
+        #
+        #  self.logger.warn(
+        #     "%s deprecated, will be removed in future release in favor of pbench-clear-results",
+        #     self.name,
+        # )
+
+        server = self.config.agent.parser.get("results", "server_rest_url")
+        headers = {"Content-Type": "application/json", "Authorization": None}
+        payload = {
+            "username": self.context.username,
+            "password": self.context.password,
+            "token_duration": self.context.token_duration
+        }
+        try:
+            response = requests.post(
+                f"{server}/login", headers=headers, json=payload
+            )
+        except requests.exceptions.ConnectionError as exc:
+            raise exc.args[0].reason
+
+        payload = json.loads(response.text)
```

Comment:
You can use `response.json()` here instead of calling the `json` module explicitly, which feels a little cleaner.

Reply:
Nice!  Done.